### PR TITLE
Update EntityCache class

### PR DESCRIPTION
**Pull request by darecoder1999**
Added `getMultipleDocuments` method

### CODE REVIEW

File: extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java [hunk 1]
```diff
@@ -77,6 +80,13 @@ public static EntityCache getEntityCache() {
         return _entityCache;
     }
 
+    public Stream<EntityDocument> getMultipleDocuments(List<EntityIdValue> entityIds) throws ExecutionException {
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java:83]: Why `Stream` and not `List`?

File: extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java [hunk 2]
```diff
@@ -77,6 +80,13 @@ public static EntityCache getEntityCache() {
...
+        List<String> ids = new ArrayList<>();
+        for (EntityIdValue entityId : entityIds) {
+            ids.add(entityId.getId());
+        }
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java:87]: You could probably use the streaming API for this instead: `entityIds.stream().map(…).collect(…)`

**Comment by darecoder1999** [comment]
Earlier, I made changes over [here](https://github.com/OpenRefine/OpenRefine/blob/master/extensions/wikidata/src/org/openrefine/wikidata/utils/EntityCache.java#L55) and changed it `new CacheLoader<List<String>, Map<String, EntityDocument>>()` and `load` method as `public Map<String, EntityDocument> load(List<String> entityIds)` and so on the other dependent files. Can't be this a solution?

Then, we don't have to use `getEntityDocument` else we will have to use `getEntityDocuments`

**Comment by wetneb** [comment]
No, this would not work. Consider the following scenario

```
EntityCache cache = new EntityCache();
cache.get(Arrays.asList(idA, idB)); // this triggers one HTTP request to Wikidata: fine
cache.get(Arrays.asList(idC, idD)); // this triggers another HTTP request to Wikidata, since these ids are new: fine again
cache.get(Arrays.asList(idB, idC)); // this will trigger another HTTP request again, because the list "idB, idC" has never been requested before, even if those documents are already in the cache! This is bad.
```

**Comment by wetneb** [comment]
Steps for next week (minutes from our meeting):
* Fix the `loadAll` method signature
* Add a test which loads multiple ids at once in `EntityCacheTests` (feel free to reuse the scenario above)
* Introduce a class to extract the set of `PropertyIdValue` mentioned in a `WikibaseSchema`; add tests for it
* Wire up this property extractor as follows:
   * before inspecting a batch, extract all the property id values that appear in the schema;
   * prefetch these properties with EntityCache
   * run scrutinizers as usual

**Pull request merged**